.ibm/pipelines/openshift-ci-tests.sh

```diff
@@ -23,496 +23,72 @@ cleanup() {
-  for resource_type in "pipelineruns.tekton.dev" "taskruns.tekton.dev"; do
-    for resource in $(oc get "$resource_type" -n "$project" -o name); do
-      oc patch "$resource" -n "$project" --type='merge' -p '{"metadata":{"finalizers":[]}}' || true
-      echo "Removed finalizers from $resource in $project."
-    done
-  done
-
-  # Check and remove specific finalizers stuck on 'chains.tekton.dev' resources
-  for chain_resource in $(oc get pipelineruns.tekton.dev,taskruns.tekton.dev -n "$project" -o name); do
-    oc patch "$chain_resource" -n "$project" --type='json' -p='[{"op": "remove", "path": "/metadata/finalizers"}]' || true
-    echo "Removed Tekton finalizers from $chain_resource in $project."
-  done
-}
-
-# Function to forcibly delete a namespace stuck in 'Terminating' status
-force_delete_namespace() {
-  local project=$1
-  echo "Forcefully deleting namespace ${project}."
-  oc get namespace "$project" -o json | jq '.spec = {"finalizers":[]}' | oc replace --raw "/api/v1/namespaces/$project/finalize" -f -
-}
+source "${DIR}/env_variables.sh"
+echo "Loaded env_variables.sh"
+source "${DIR}/utils.sh"
+echo "Loaded utils.sh"
+source "${DIR}/jobs/aks.sh"
+echo "Loaded aks.sh"
+source "${DIR}/jobs/gke.sh"
+echo "Loaded gke.sh"
+source "${DIR}/jobs/main.sh"
+echo "Loaded main.sh"
+source "${DIR}/jobs/ocp-v4-15.sh"
+echo "Loaded ocp-v4-15.sh"
+source "${DIR}/jobs/ocp-v4-16.sh"
+echo "Loaded ocp-v4-16.sh"
+source "${DIR}/jobs/operator.sh"
+echo "Loaded operator.sh"
+source "${DIR}/jobs/periodic.sh"
+echo "Loaded periodic.sh"
 
 main() {
   echo "Log file: ${LOGFILE}"
-  set_cluster_info
-  source "${DIR}/env_variables.sh"
-
-  install_oc
-  if [[ "$JOB_NAME" == *aks* ]]; then
-    az_login
-    az_aks_start "${AKS_NIGHTLY_CLUSTER_NAME}" "${AKS_NIGHTLY_CLUSTER_RESOURCEGROUP}"
-    az_aks_approuting_enable "${AKS_NIGHTLY_CLUSTER_NAME}" "${AKS_NIGHTLY_CLUSTER_RESOURCEGROUP}"
-    az_aks_get_credentials "${AKS_NIGHTLY_CLUSTER_NAME}" "${AKS_NIGHTLY_CLUSTER_RESOURCEGROUP}"
-  elif [[ "$JOB_NAME" == *gke* ]]; then
-    gcloud_auth "${GKE_SERVICE_ACCOUNT_NAME}" "/tmp/secrets/GKE_SERVICE_ACCOUNT_KEY"
-    gcloud_gke_get_credentials "${GKE_CLUSTER_NAME}" "${GKE_CLUSTER_REGION}" "${GOOGLE_CLOUD_PROJECT}"
-  else
-    oc login --token="${K8S_CLUSTER_TOKEN}" --server="${K8S_CLUSTER_URL}"
-  fi
-  echo "OCP version: $(oc version)"
-
-  set_namespace
 
   API_SERVER_URL=$(oc whoami --show-server)
-  if [[ "$JOB_NAME" == *aks* ]]; then
-    export K8S_CLUSTER_ROUTER_BASE=$AKS_INSTANCE_DOMAIN_NAME
-  elif [[ "$JOB_NAME" == *gke* ]]; then
-    export K8S_CLUSTER_ROUTER_BASE=$GKE_INSTANCE_DOMAIN_NAME
-  else
-    export K8S_CLUSTER_ROUTER_BASE=$(oc get route console -n openshift-console -o=jsonpath='{.spec.host}' | sed 's/^[^.]*\.//')
-  fi
-
-  echo "K8S_CLUSTER_ROUTER_BASE : $K8S_CLUSTER_ROUTER_BASE"
-
   ENCODED_API_SERVER_URL=$(echo "${API_SERVER_URL}" | base64)
   ENCODED_CLUSTER_NAME=$(echo "my-cluster" | base64)
```

Comment:
Doing `oc whoami --show-server` at this point breaks the AKS and GKE because the cluster authentication is happening in the `handle`s. I would suggest that you move this part into handles for all the OCP clusters, or perhaps create a function and reference it from the OCP handles.